seisbench/models/dkpn.py

```diff
@@ -0,0 +1,784 @@
+        :type update: bool
+        :param wait_for_file: Whether to wait on partially downloaded files, defaults to False
+        :type wait_for_file: bool, optional
+        :return: Model instance
+        :rtype: SeisBenchModel
+        """
+        cls._cleanup_local_repository()
+        _cache_migration_v0_v3()
+
+        if version_str == "latest":
+            versions = cls.list_versions(name, remote=update)
+            # Always query remote versions if cache is empty
+            if len(versions) == 0:
+                versions = cls.list_versions(name, remote=True)
+
+            if len(versions) == 0:
+                raise ValueError(f"No version for weight '{name}' available.")
+            version_str = max(versions, key=version.parse)
+
+        weight_path, metadata_path = cls._pretrained_path(name, version_str)
+
+        cls._ensure_weight_files(
+            name, version_str, weight_path, metadata_path, force, wait_for_file
+        )
+
+        if metadata_path.is_file():
+            with open(metadata_path, "r") as f:
+                weights_metadata = json.load(f)
+        else:
+            weights_metadata = {}
+        model_args = weights_metadata.get("model_args", {})
+        model_args["in_channels"] = 4
+        model = cls(**model_args)
+
+        model._weights_metadata = weights_metadata
+        model._parse_metadata()
+
+        state_dict = torch.load(weight_path)
+        old_weight = state_dict["inc.weight"]
+        state_dict["inc.weight"] = torch.zeros(
+            old_weight.shape[0], old_weight.shape[1] + 1, old_weight.shape[2]
+        ).type_as(old_weight)
+        state_dict["inc.weight"][:, :3, ...] = old_weight
+        state_dict["inc.weight"][:, 3, ...] = old_weight[:, 0, ...]
+        model.load_state_dict(state_dict)
+        return model
+
+
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================  PREPROC
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+
+
+class PreProc(object):
+    def __init__(self, **kwargs):
+        self.stream = None  # --> Instantiated everytime in the work() class-method
```

Comment:
As for the DKPN class, temporary variables should not be saved in class variables. Instead, you could make `work` return the processed stream.